lib/WUI/nhttp/file_info.cpp

```diff
@@ -123,7 +125,7 @@ FileInfo::FileInfo(const char *filename, bool can_keep_alive, bool json_errors,
 Step FileInfo::step(std::string_view, bool, uint8_t *output, size_t output_size) {
     const size_t last_chunk_len = strlen(LAST_CHUNK);
     size_t written = 0;
-    const ConnectionHandling handling = can_keep_alive ? ConnectionHandling::ChunkedKeep : ConnectionHandling::Close;
+    const ConnectionHandling handling = can_keep_alive && !header_only ? ConnectionHandling::ChunkedKeep : ConnectionHandling::Close;
```

Comment:
Why do you disable keep-alive on the header_only? That probably isn't desirable.